ios/rxswift_sherpa.md

```diff
@@ -1,4 +1,122 @@
 # Rx{Swift,Cocoa} Sherpa
+
+*** **This document is under active development** ***
+
+This document contains best practices when using RxSwift, RxCocoa, and other Rx libraries in iOS projects.
+
+## Rx-enabled View Models
+
+### Setting up your Observables
+
+When possible, a ViewModel's observable sequences should be setup in a declarative fashion within the ViewModel's `init` method or in lazily declared properties. This allows you to specify your exposed observables as non-optional immutable `let` properties.
+
+```
+private let user = Variable<User?>(nil)
+let username: Observable<String>
+
+init(identifier: String) {
+  APIClient.user(withIdentifier: identifier).catchErrorJustReturn(nil).bind(to: user).disposed(by: disposeBag)
+  username = user.asObservable.map({ $0.username })
+}
+```
+
+The majority of a ViewModel's exposed observable sequences will be observed by objects in the UI layer. CocoaTouch requires that UI updates be performed on the main thread. Additionally, since these observations should generally continue until the ViewController leaves the screen, the observed sequences should not complete or error. For these reasons, you should use the correct observable type for the given situation:
+
+* Use a `Driver` when data is set by the ViewModel and is read-only from the View's perspective
+* Use a `Variable` when double-binding is required to allow the observing View to update observable sequence's most recent value based on user input
+* Use of `Observable` should be limited to when the initial stream is a immediately available; although, in these cases, a plain swift property may be sufficient
+
+__Open Question: Should we be advocating against using Observable at all?__
```

Comment:
If we're talking about in the context of a view model exposing observable sequences to a view controller then yeah, I'd advocate against using `Observable`s at all. Not in general tho; for instance in the example above `APIClient.user(_:)` should absolutely return a normal `Observable`, but `username` should be a `Driver`.

Reply:
It's probably clearer if we just clarify what a `Driver` is (namely, `Observable` + `.observeOn(MainScheduler.instance)` + `.shareReplayLatestWhileConnected()`), then stipulate that only `Driver`s should be bound to UI elements